pipelines/toast_ground_sim.py

```diff
@@ -547,415 +162,173 @@ def parse_arguments(comm):
-    schedules = comm.comm_world.bcast(schedules)
-
-    tmr.stop()
-    if comm.world_rank == 0:
-        tmr.report("Loading schedule")
-    return schedules
-
-
-@function_timer
-def get_focalplane_radius(args, focalplane, rmin=1.0):
-    """ Find the furthest angular distance from the boresight
-
-    """
-    if args.focalplane_radius:
-        return args.focalplane_radius
-
-    cosangs = []
-    for det in focalplane:
-        quat = focalplane[det]["quat"]
-        vec = qa.rotate(quat, ZAXIS)
-        cosangs.append(np.dot(ZAXIS, vec))
-    mincos = np.amin(cosangs)
-    maxdist = max(np.degrees(np.arccos(mincos)), rmin)
-    return maxdist * 1.001
+    return args, comm
 
 
 @function_timer
 def load_focalplanes(args, comm, schedules):
     """ Attach a focalplane to each of the schedules.
 
+    Args:
+        schedules (list) :  List of tuples of the form
+            (`site`, `all_ces`) where `site` is a Site
+            instansce with the `telescope` field filled
+            and `all_ces` is a list of CES objects.
+    Returns:
+        detweights (dict) : Inverse variance noise weights for every
+            detector across all focal planes.
     """
-    tmr = Timer()
-    tmr.start()
+    timer = Timer()
+    timer.start()
 
+    telescopes = []
+    for schedule in schedules:
+        telescopes.append(schedule[0].telescope)
     # Load focalplane information
 
     focalplanes = []
     if comm.world_rank == 0:
-        ftmr = Timer()
-        for fpfile in args.fp.split(","):
-            ftmr.start()
+        ftimer = Timer()
+        for fpfile in args.focalplane.split(","):
+            ftimer.start()
             with open(fpfile, "rb") as picklefile:
                 focalplane = pickle.load(picklefile)
                 focalplanes.append(focalplane)
-                ftmr.report_clear("Load {}".format(fpfile))
-        ftmr.stop()
-    focalplanes = comm.comm_world.bcast(focalplanes)
+                ftimer.report_clear("Load {}".format(fpfile))
+        ftimer.stop()
+    if comm.comm_world is not None:
+        focalplanes = comm.comm_world.bcast(focalplanes)
+        telescopes = comm.comm_world.bcast(telescopes)
 
     if len(focalplanes) == 1 and len(schedules) > 1:
         focalplanes *= len(schedules)
+    if len(telescopes) == 1 and len(schedules) > 1:
+        telescopes *= len(schedules)
     if len(focalplanes) != len(schedules):
         raise RuntimeError(
             "Number of focalplanes must equal number of schedules or be 1."
         )
 
+    # Append a focal plane and telescope to each entry in the schedules list
     detweights = {}
-    for schedule, focalplane in zip(schedules, focalplanes):
+    for schedule, focalplane, telescope in zip(schedules, focalplanes, telescopes):
         schedule.append(focalplane)
-        for detname, det in focalplane.items():
-            net = det["NET"]
-            detweight = 1.0 / (args.samplerate * net * net)
+        schedule.append(Telescope(telescope))
```

Comment:
`schedule` is made up of both `focalplane` and `Telescope` objects?

Reply:
interleaved?

Reply:
Telescope is just a short-hand to store the name and ID of a telescope. I think TOAST would benefit from a fully-fledged Telescope class (with Focalplane as a member) but that is beyond the scope of the pipeline refactor.